### ISSUE TITLE
Consider revising shadow on navigation to better align with our shadow usage guidelines

### ISSUE THREAD

**Issue by mceledonia** [patternfly/patternfly-design#1164]
Currently our navigation component has a drop shadow behind it. This was carried into dark theme where it is more prominent due to being used as a visual separator between the navigation and the page.

We previously established rules around shadow usage - that they should only be used on elements overlapping other content. For example toast alerts, dropdown menus, and popovers.

The navigation has always been an exception to this rule due to being grandfathered in prior to the rules being established. However there is no need for the shadow on default theme due to the dark background color providing sufficient contrast from the rest of the page. On dark theme the shadow would be more appropriate as a standard border due to being an inline element on the page. 

TL;DR - Proposing:
- Remove shadow from navigation on default theme
- Replace shadow on navigation in dark theme with standard border

This would not be a breaking change as it isn't changing spacing or sizing, just shadow/border treatment.

CC @mcarrano @mcoker @mmenestr

**Comment by mceledonia** [comment]
I found a mockup from my dark theme working file which uses a border for the nav so it is likely this slipped through the cracks:
![image](https://user-images.githubusercontent.com/30935938/170320343-efa3aba1-e7fc-47d1-b0af-50bf5adeee79.png)

**Comment by mmenestr** [comment]
I think that's fine if you're planning on keeping the navigation the same color as the header background.. though in general, I wonder if we should make the navigation darker so there's more contrast between the header and nav? But lighter than the black area surrounding the table

**Comment by mcoker** [comment]
@mcarrano @mceledonia nope, should be fine. Here are some screenshots. Though we do still support a "light" variant of the nav - would we leave the shadow or use a border there, too?

### Screenshots of this update
<img width="2322" alt="Screen Shot 2022-06-17 at 9 56 50 AM" src="https://user-images.githubusercontent.com/35148959/174324312-c55dc1dd-af27-4293-8c32-653dfcb7510a.png">

<img width="2322" alt="Screen Shot 2022-06-17 at 9 56 45 AM" src="https://user-images.githubusercontent.com/35148959/174324314-f9a3094a-bcff-4c4c-aeaf-991241d45cc2.png">

### Light - shadow (existing)
<img width="2322" alt="Screen Shot 2022-06-17 at 9 59 29 AM" src="https://user-images.githubusercontent.com/35148959/174324309-05117405-d746-4d20-b09f-d967eacef6f9.png">

### Light - border
<img width="2322" alt="Screen Shot 2022-06-17 at 9 59 58 AM" src="https://user-images.githubusercontent.com/35148959/174324291-2aa9df49-439e-4ce9-bc6d-2afef431d5a1.png">

**Comment by mcarrano** [comment]
I agree, also. I have opened core issue https://github.com/patternfly/patternfly/issues/4925 to perform this update.

@doruskova does this require a design kit update?

**Comment by doruskova** [comment]
@mcarrano Yes. I opened the issue for it https://github.com/patternfly/patternfly-design-kit/issues/557
